src/cockpit/packages.py

```diff
@@ -57,31 +57,24 @@ def directory_items(path):
 # - rewrite cockpit-askpass in Python and write it into a temporary file
 # - bundle helper shell scripts into their webpacks
 # Until then, we need this libexecdir detection hack.
-LIBEXECDIR = None
-
-
-def get_libexecdir() -> str:
-    '''Detect libexecdir on current machine
-
-    This only works for systems which have cockpit-ws installed.
-    '''
-    global LIBEXECDIR
-    if LIBEXECDIR is None:
-        for candidate in ['/usr/local/libexec', '/usr/libexec', '/usr/local/lib/cockpit', '/usr/lib/cockpit']:
-            if os.path.exists(os.path.join(candidate, 'cockpit-askpass')):
-                LIBEXECDIR = candidate
-                break
-        else:
-            logger.warning('Could not detect libexecdir')
-            # give readable error messages
-            LIBEXECDIR = '/nonexistent/libexec'
-
-    return LIBEXECDIR
+LIBEXEC_PATH = [
+    '/usr/local/libexec',
+    '/usr/libexec',
+    '/usr/local/lib/cockpit',
+    '/usr/lib/cockpit',
+    *os.environ['PATH'].split(':')
```

Comment:
Not a fan of adding `$PATH` here, but then that whole `${libexecdir}` replacing is only temporary anyway. :grin:  But see the previous comment, we can fix this rather easily.